sale_commission/model/sale_order.py

```diff
@@ -0,0 +1,218 @@
+        "sale.order.agent",
+        "sale_id",
+        string="Agents",
+        states={"draft": [("readonly", False)]}
+    )
+
+    # XXX: It doesn't work with Odoo v8.0 and I think it is not necessary
+    # def create(self, cr, uid, values, context=None):
+    #     """Add sale order reference on sale.order.agent
+    #     """
+    #     if context is None:
+    #         context = {}
+    #     agent_pool = self.pool.get('sale.order.agent')
+    #     res = super(sale_order, self).create(
+    #          cr, uid, values, context=context)
+    #     if 'sale_agent_ids' in values:
+    #         for sale_order_agent in values['sale_agent_ids']:
+    #             agent_pool.write(
+    #           cr, uid, sale_order_agent[1], {'sale_id': res})
+    #     return res
+
+    # XXX: It doesn't work with Odoo v8.0 and I think it is not necessary
+    # def write(self, cr, uid, ids, values, context=None):
+    #     """Rebuild sale.order reference on sale.order.agent
+    #     """
+    #     if context is None:
+    #         context = {}
+    #     agent_pool = self.pool.get('sale.order.agent')
+    #     if 'sale_agent_ids' in values:
+    #         for sale_order_agent in values['sale_agent_ids']:
+    #             for id in ids:
+    #                 if sale_order_agent[2]:
+    #                     sale_order_agent[2]['sale_id'] = id
+    #                 else:
+    #                     agent_pool.unlink(
+    #           cr, uid, sale_order_agent[1], context=context)
+    #     return super(sale_order, self).write(
+    #           cr, uid, ids, values, context=context)
+
+    def onchange_partner_id(self, cr, uid, ids, part, context=None):
+        """Agent id field will be changed according to new partner
+        """
+        if context is None:
+            context = {}
+        sale_agent_ids = []
+        res = super(sale_order, self).onchange_partner_id(
+            cr, uid, ids, part, context=context
+        )
+        if res.get('value', False) and part:
+            sale_order_agent = self.pool.get('sale.order.agent')
+            if ids:
+                agent_id = sale_order_agent.search(
+                    cr, uid,
+                    [('sale_id', '=', ids)],
+                    context=context
+                )
+                sale_order_agent.unlink(cr, uid, agent_id, context=context)
+            partner_obj = self.pool.get('res.partner')
+            partner = partner_obj.browse(cr, uid, part, context=context)
+            for partner_agent in partner.commission_ids:
+                vals = {
+                    'agent_id': partner_agent.agent_id.id,
+                    'commission_id': partner_agent.commission_id.id,
+                    # 'sale_id':ids
+                }
+                # FIXME: What is going on in this block?
+                if ids:
+                    for id in ids:
+                        vals['sale_id'] = id
+                sale_agent_id = sale_order_agent.create(
```

Comment:
I guess there is an error of indentation here.

We certainly don't want to create all `sale.order.agent` with `sale_id = ids[-1]`